osquery/core/tests/watcher_tests.cpp

```diff
@@ -261,4 +267,41 @@ TEST_F(WatcherTests, test_watcherrunner_watcherhealth) {
   runner.isWatcherHealthy(*test_process, state);
   EXPECT_EQ(2U, state.sustained_latency);
 }
+
+TEST_F(WatcherTests, test_watcherrunner_unhealthy_delay) {
+  FakeWatcherRunner runner(0, nullptr, true);
+
+  auto test_process = PlatformProcess::getCurrentProcess();
+  auto fake_test_process = FakePlatformProcess(test_process->nativeHandle());
+  fake_test_process.setStatus(PROCESS_STILL_ALIVE, 0);
+
+  // Set up a fake test process and place it into an unhealthy state.
+  Row r;
+  r["parent"] = INTEGER(test_process->nativeHandle());
```

Comment:
Ah man, here's the issue, amongst another thing. Handles on Windows aren't `pid_t` values, rather they're Windows `HANDLE` values. I tried pushing a commit to your branch, not sure it worked or not, but if you change this to be something like:
```
  r["parent"] = isPlatform(PlatformType::TYPE_WINDOWS)
                    ? INTEGER(test_process->pid())
                    : INTEGER(test_process->nativeHandle());
```
It'll work without additional changes. 

The exception was a bug with the Watchdog handling it's workers and we had an Access violation on a null reference. You just so happened to hit it right and get the read violation to trigger, but making the above changes side-steps it. I can either finagle a way to get the commit (hash 414fdd67e393d557daccd2be23f32a831690ad5f), or send you a patch or throw up another PR fixing the bug